django/db/backends/mysql/operations.py

```diff
@@ -39,6 +39,10 @@ def date_trunc_sql(self, lookup_type, field_name):
         if lookup_type in fields:
             format_str = fields[lookup_type]
             return "CAST(DATE_FORMAT(%s, '%s') AS DATE)" % (field_name, format_str)
+        elif lookup_type == 'quarter':
+            return "MAKEDATE(YEAR(%s), 1) + INTERVAL QUARTER(%s) QUARTER - INTERVAL 1 QUARTER" % (
```

Comment:
can't see anything obvious after refamiliarizing myself with the [date/time functions](https://mariadb.com/kb/en/mariadb/date-and-time-functions/)